src/dispatch/data/source/service.py

```diff
@@ -0,0 +1,253 @@
+from dispatch.data.source.type import service as type_service
+from dispatch.data.source.transport import service as transport_service
+
+
+from .models import Source, SourceCreate, SourceUpdate, SourceRead
+
+
+def get(*, db_session, source_id: int) -> Optional[Source]:
+    """Gets a source by its id."""
+    return db_session.query(Source).filter(Source.id == source_id).one_or_none()
+
+
+def get_by_name(*, db_session, project_id: int, name: str) -> Optional[Source]:
+    """Gets a source by its name."""
+    return (
+        db_session.query(Source)
+        .filter(Source.name == name)
+        .filter(Source.project_id == project_id)
+        .one_or_none()
+    )
+
+
+def get_by_name_or_raise(*, db_session, project_id, source_in: SourceRead) -> SourceRead:
+    """Returns the source specified or raises ValidationError."""
+    source = get_by_name(db_session=db_session, project_id=project_id, name=source_in.name)
+
+    if not source:
+        raise ValidationError(
+            [
+                ErrorWrapper(
+                    NotFoundError(
+                        msg="Source not found.",
+                        source=source_in.name,
+                    ),
+                    loc="source",
+                )
+            ],
+            model=SourceRead,
+        )
+
+    return source
+
+
+def get_all(*, db_session, project_id: int) -> List[Optional[Source]]:
+    """Gets all sources."""
+    return db_session.query(Source).filter(Source.project_id == project_id)
+
+
+def create(*, db_session, source_in: SourceCreate) -> Source:
+    """Creates a new source."""
+    project = project_service.get_by_name_or_raise(
+        db_session=db_session, project_in=source_in.project
+    )
+
+    source = Source(
+        **source_in.dict(
+            exclude={
+                "project",
+                "owner",
+                "tags",
+                "source_environment",
+                "source_data_format",
+                "source_transport",
+                "source_status",
+                "source_type",
+            }
+        ),
+        project=project,
+    )
+
+    if source_in.owner:
+        source.source_owner = service_service.get_by_name_or_raise(
+            db_session=db_session, project_id=project.id, service_in=source_in.owner
+        )
+
+    tags = []
+    for t in source_in.tags:
+        tags.append(
+            tag_service.get_by_name_or_raise(db_session=db_session, tag_in=t, project_id=project.id)
+        )
+
+    source.tags = tags
+
+    incidents = []
+    for i in source_in.incidents:
+        incidents.append(
+            incident_service.get_by_name_or_raise(
+                db_session=db_session, incident_in=i, project_id=project.id
+            )
+        )
+
+    queries = []
+    for q in source_in.queries:
+        queries.append(
+            query_service.get_by_name_or_raise(
+                db_session=db_session, query_in=q, project_id=project.id
+            )
+        )
+
```

Comment:
Are we missing `source.queries = queries` here? I don't see a relationship for queries in the data model.